src/containers/exchange/components/SimpleBuyForm.js

```diff
@@ -0,0 +1,351 @@
+import { AEUR, ETH } from "components/augmint-ui/currencies.js";
+import { Tokens } from "@augmint/js";
+
+import theme from "styles/theme";
+import styled from "styled-components";
+import { SIMPLE_BUY_SUCCESS } from "../../../modules/reducers/orders";
+
+const Styledlabel = styled.label`
+    display: inline-block;
+    margin-bottom: 5px;
+`;
+
+const StyledBox = styled.div`
+    border-radius: 3px;
+    box-sizing: border-box;
+    width: 100%;
+    padding: 15px;
+    margin-bottom: 20px;
+    background: rgba(232, 232, 232, 0.3);
+    border: 2px solid rgba(232, 232, 232, 0.3);
+
+    text-align: center;
+    font-size: 14px;
+    line-height: 200%;
+    color: rgba(0, 0, 0, 0.7);
+
+    & strong {
+        font-size: 16px;
+    }
+
+    &.validation-error {
+        border: 2px solid ${theme.colors.darkRed};
+        background-color: ${theme.colors.lightRed};
+        & strong.err {
+            color: ${theme.colors.darkRed};
+        }
+    }
+`;
+
+class SimpleBuyForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            simpleResult: null,
+            result: null,
+            orderDirection: TOKEN_BUY,
+            liquidityError: null,
+            averagePrice: null
+        };
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.onOrderDirectionChange = this.onOrderDirectionChange.bind(this);
+        this.onTokenAmountChange = this.onTokenAmountChange.bind(this);
+    }
+
+    onOrderDirectionChange(e) {
+        const orderDirection = +e.target.attributes["data-index"].value;
+
+        this.setState({
+            orderDirection
+        });
+        this.onTokenAmountChange(null, orderDirection, this.state.inputVal);
+    }
+
+    async getSimpleResult(token) {
+        const { ethFiatRate } = this.props.rates.info;
+        const bn_ethFiatRate = ethFiatRate !== null && new BigNumber(ethFiatRate);
+        const isBuy = this.state.orderDirection === TOKEN_BUY;
+
+        const res = await store.dispatch(getSimpleBuy(token, isBuy, bn_ethFiatRate));
+
+        if (res.type !== SIMPLE_BUY_SUCCESS) {
+            throw new SubmissionError({
+                _error: res.error
+            });
+        } else {
+            const averagePrice = this.props.rates.info.ethFiatRate / res.result.averagePrice.toNumber();
+            let liquidityError = null;
+
+            if (this.state.orderDirection === TOKEN_BUY) {
+                liquidityError = this.maxExchangeValue(token, res.result.filledTokens);
+            }
+
+            this.setState({
+                simpleResult: res.result,
+                averagePrice: averagePrice,
+                liquidityError: liquidityError
+            });
+            return;
+        }
+    }
+
+    onTokenAmountChange(e, direction, savedValue) {
+        const value = e ? e.target.value : savedValue;
+        const simpleResult = value > 0 ? this.getSimpleResult(parseFloat(value)) : null;
```

Comment:
the getSimpleResult always returns undefined, so simpleResult will always be a Promise<undefined>